app/components/learningmaterial-search.js

```diff
@@ -1,99 +1,101 @@
 import Ember from 'ember';
 
-const { Component, computed, inject, run } = Ember;
+const { Component, inject } = Ember;
 const { service } = inject;
-const { debounce } = run;
-const { notEmpty, or } = computed;
 
-export default Component.extend({
-  classNames: ['learningmaterial-search'],
+var ProxiedMaterials = Ember.ObjectProxy.extend({
+  isActive: false
+});
 
+export default Component.extend({
   store: service(),
   i18n: service(),
-
-  results: [],
+  classNames: ['learningmaterial-search'],
   currentMaterials: [],
+  query: '',
+  searchResults: [],
+  searchPage: 0,
+  searchResultsPerPage: 50,
+  hasMoreSearchResults: false,
+  targetItemTitle: '',
   searching: false,
-  showMoreInputPrompt: false,
-  showNoResultsMessage: false,
-  currentlySearchingForTerm: false,
-
-  hasResults: notEmpty('results'),
-
-  showList: or('searching', 'showMoreInputPrompt', 'showNoResultsMessage', 'hasResults'),
+  searchReturned: false,
 
   actions: {
-    clear() {
-      let input = this.$('input')[0];
-      input.value = '';
-
-      this.setProperties({
-        searchTerms: '',
-        showMoreInputPrompt: false,
-        showNoResultsMessage: false,
-        searching: false,
-        results: [],
-        showClearButton: false,
-      });
-    },
-
-    inputValueChanged() {
-      let input = this.$('input')[0];
-      let searchTerms = input.value;
-
-      if (this.get('currentlySearchingForTerm') === searchTerms) {
+    search: function(query){
```

Comment:
This should be converted to a task to mitigate annoyances that happen when a promise returns after a component is removed from the dom.